data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/metrics/OpenSearchSourcePluginMetrics.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.opensearch.metrics;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Timer;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+
+public class OpenSearchSourcePluginMetrics {
+
+    static final String DOCUMENTS_PROCESSED = "documentsProcessed";
+    static final String INDICES_PROCESSED = "indicesProcessed";
+    static final String INDEX_PROCESSING_TIME_ELAPSED = "indexProcessingTime";
+    static final String PROCESSING_ERRORS = "processingErrors";
+
+    private final Counter documentsProcessedCounter;
+    private final Counter indicesProcessedCounter;
+    private final Counter processingErrorsCounter;
+    private final Timer indexProcessingTimeTimer;
+
+    public static OpenSearchSourcePluginMetrics create(final PluginMetrics pluginMetrics) {
+        return new OpenSearchSourcePluginMetrics(pluginMetrics);
+    }
+
+    private OpenSearchSourcePluginMetrics(final PluginMetrics pluginMetrics) {
+        documentsProcessedCounter = pluginMetrics.counter(DOCUMENTS_PROCESSED);
+        indicesProcessedCounter = pluginMetrics.counter(INDICES_PROCESSED);
+        processingErrorsCounter = pluginMetrics.counter(PROCESSING_ERRORS);
+        indexProcessingTimeTimer = pluginMetrics.timer(INDEX_PROCESSING_TIME_ELAPSED);
+    }
+
+    public Counter getDocumentsProcessedCounter() {
```

Comment:
An alternative design would be to increment from this class itself.

```
public void incrementDocumentsProcessedCounter()
```

Also, compare to the Law of Demeter - https://en.wikipedia.org/wiki/Law_of_Demeter

I'm ok with this either way though.
From the perspective of this class, reading the counters isn't really something it is trying to provide. It expects that its caller is just incrementing.